app-agent/src/exec_process.rs

```diff
@@ -12,15 +17,33 @@ use alumet::{
     plugin::event::StartConsumerMeasurement,
     resources::ResourceConsumer,
 };
-use anyhow::Context;
+use anyhow::{anyhow, Context};
 
 /// Spawns a child process and waits for it to exit.
 pub fn exec_child(external_command: String, args: Vec<String>) -> anyhow::Result<ExitStatus> {
     // Spawn the process.
-    let mut p = std::process::Command::new(external_command.clone())
-        .args(args)
-        .spawn()
-        .expect("error in child process");
+    let p_result = std::process::Command::new(external_command.clone())
+        .args(args.clone())
+        .spawn();
+
+    let mut p = match p_result {
+        Ok(val) => val,
+        Err(e) => match e.kind() {
+            std::io::ErrorKind::NotFound => {
+                let return_error: String = handle_not_found(external_command, args);
+                log::error!("{}", return_error);
+                return Err(e.into());
```

Comment:
The error should be logged when it is handled, not here IMO. (=> remove log::error)